public/docs/dart/latest/tutorial/toh-pt3.jade

```diff
@@ -1,14 +1,251 @@
-    A preliminary Dart version of the Tour of Heroes app,
-    featuring the hero editor, a master/detail page,
-    multiple components, services, and routing
+.l-main-section
+:marked
+  ## Where We Left Off
+  Before we continue with our Tour of Heroes, let’s verify we have the following structure. If not, we’ll need to go back and follow the previous chapters.
+
+.filetree
+  .file angular2_tour_of_heroes
+  .children
+    .file lib
+    .children
+      .file app_component.dart
+    .file web
+    .children
+      .file index.html
+      .file main.dart
+    .file pubspec.yaml
+:marked
+  ### Keep the app compiling and running
+  We want to start the Dart compiler, have it watch for changes, and start our server. We'll do this by typing
+
+code-example(format="." language="bash").
+  pub serve
+
+:marked
+  This will keep the application running while we continue to build the Tour of Heroes.
+
+  ## Making a Hero Detail Component
+  Our heroes list and our hero details are in the same component in the same file.
+  They're small now but each could grow.
+  We are sure to receive new requirements for one and not the other.
+  Yet every change puts both components at risk and doubles the testing burden without benefit.
+  If we had to reuse the hero details elsewhere in our app,
+  the heroes list would tag along for the ride.
+
+  Our current component violates the
+  [Single Responsibility Principle](https://blog.8thlight.com/uncle-bob/2014/05/08/SingleReponsibilityPrinciple.html).
+  It's only a tutorial but we can still do things right &mdash;
+  especially if doing them right is easy and we learn how to build Angular apps in the process.
+
+  Let’s break the hero details out into its own component.
+
+  ### Separating the Hero Detail Component
+  Add a new file named `hero_detail_component.dart` to the `lib` folder and create `HeroDetailComponent` as follows.
+
++makeExample('toh-3/dart/lib/hero_detail_component.dart', 'v1', 'hero_detail_component.dart (initial version)')(format=".")
+.l-sub-section
+  :marked
+    ### Naming conventions
+    We like to identify at a glance which classes are components and which files contain components.
+
+    Notice that  we have an `AppComponent` in a file named `app_component.dart` and our new
+    `HeroDetailComponent` is in a file named `hero_detail_component.dart`.
+
+    All of our component names end in "Component".  All of our component file names end in "_component".
+
+    We spell our file names in lower dash case (AKA "kebab case") so we don't worry about
+    case sensitivity on the server or in source control.
+
+    <!-- TODO
+    .l-sub-section
+      :marked
+        Learn more about naming conventions in the chapter [Naming Conventions]
+    :marked
+    -->
+:marked
+  We begin by importing the Angular `core.dart` file,
+  so that we can use common types like `@Component` when we create
```

Comment:
Hmm, the type is actually `Component` instead of `Component`. If you use it with `@` it would probably be better to use `@Component()` (with `()`)